### PR TITLE
♻ prepare for production

### PR DESCRIPTION
**Pull request by Ked57**
I did a few things

- Add a docker-compose file for local testing
- Change the method to start the app from PM2 to a more conventional method
- Reorganise the files
- Updade theconfig file so it works with env variables

### CODE REVIEW

File: src/models/index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const Sequelize = require('sequelize');
 
 const basename = path.basename(__filename);
 const env = process.env.NODE_ENV || 'development';
-const config = require(`${__dirname}/../config/config.json`)[env];
+const config = require(`${__dirname}/../../config/config.js`)[env];
```
Comment (hgwood) [inline src/models/index.js]:
```suggestion
const config = require(`${__dirname}/../../config/config`)[env];
```

File: package.json [hunk 2]
```diff
@@ -6,9 +6,11 @@
   "author": "",
   "license": "ISC",
   "scripts": {
-    "start": "pm2 start ./bin/server --watch",
-    "kill": "pm2 kill",
-    "log": "pm2 logs",
+    "start": "node ./bin/server",
```
Comment (hgwood) [inline package.json]: `./bin/server` has been deleted (see also `pm2:start`)

File: package.json [hunk 3]
```diff
@@ -6,9 +6,11 @@
...
+    "dev": "nodemon --watch ./src --exec node -r dotenv/config ./src/app.js",
```
Comment (hgwood) [inline package.json]:
```suggestion
    "dev": "nodemon --watch ./src --exec node --require dotenv/config ./src/app.js",
```

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+version: "3"
+services:
+  postgres:
+    image: mysql:latest
+    ports: 
+      - "3306:3306"
+    environment:
+      MYSQL_DATABASE: zintranet
+      MYSQL_ROOT_PASSWORD: root
```
Comment (hgwood) [inline docker-compose.yaml]: We could use the same env vars as in the app here. Docker Compose automatically reads `.env`. That would ensure the database container and app are always in sync for the db config.

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+version: "3"
+services:
+  postgres:
+    image: mysql:latest
```
Comment (hgwood) [inline docker-compose.yaml]: This should at least have the major version pinned down.

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+version: "3"
+services:
+  postgres:
```
Comment (hgwood) [inline docker-compose.yaml]:
```suggestion
  mysql:
```

File: config/config.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+module.exports = {
+  development: {
+    database: process.env.MYSQL_ADDON_DB,
+    username: process.env.MYSQL_ADDON_USER,
+    password: process.env.MYSQL_ADDON_PASSWORD,
+    host: process.env.MYSQL_ADDON_HOST,
+    port: process.env.MYSQL_ADDON_PORT,
+    dialect: 'mysql',
+  },
```
Comment (hgwood) [inline config/config.js]: Shouldn't this also have a production config?

File: docker-compose.yaml (hunk 4 above)
Comment (Ked57) [reply docker-compose.yaml]: I've been wondering about this, clever-cloud addons expose the MYSQL_ADDON_* env variables and the docker-compose file uses MYSQL_* env variables. I'm not sure how to do this then

**Pull request merged**